### ISSUE TITLE
sample_flux/sample_energy_flux differences and similarities

### ISSUE THREAD

**Issue by DougBurke** [sherpa/sherpa#753]
We should review these two, in particular for what they return. As an example I think `sample_energy_flux` should return the best-fit statistic along with the flux + pars array. This would simplify `sample_flux` and provide somewhat-consistent data from the two of them.

**Comment by DougBurke** [comment]
See #322

**Comment by anetasie** [comment]
@DougBurke I agree that both need to be reviewed and tested to see the differences. At this moment I would not drop `sample_flux`.
